syntactic-paths/src/test/java/com/palantir/util/syntacticpath/PathTest.java

```diff
@@ -287,4 +288,14 @@ public void testToStringAndSerialization() throws IOException {
             assertThat(mapper.readValue("\"" + path + "\"", Path.class), is(new Path(path)));
         }
     }
+
+    @Test
+    public void testGetSegments() throws Exception {
+        assertThat(new Path("/a/b").getSegments(), contains("a", "b"));
+        assertThat(new Path("a/b").getSegments(), contains("a", "b"));
+        assertThat(new Path("a/b/").getSegments(), contains("a", "b"));
+        assertThat(new Path("a//b").getSegments(), contains("a", "b"));
```

Comment:
Yep, I get that, but it's not obvious to me that this is the correct normalization.